animal_crossing.py

```diff
@@ -0,0 +1,193 @@
+from datetime import datetime, time, timezone
+import re
+import sqlite3
+import dateutil
+import dateutil.tz
+
+import config
+
+if config.bot.acnh_db is not None:
+	db = sqlite3.connect(config.bot.acnh_db)
+	db.row_factory = sqlite3.Row
+
+	# enable foreign key constraints
+	with db:
+		db.execute('PRAGMA foreign_keys = ON')
+
+friend_code_regex = re.compile(r'SW-\d{4}-\d{4}-\d{4}')
+cached_row_id = 0
+
+def stalk_market(cmd):
+	if not cmd.args:
+		# TODO: usage string? some other default behavior?
+		return
+
+	split = cmd.args.split(' ', 1)
+	subcmd = split[0]
+	if subcmd == 'tz':
+		_stalk_set_timezone(cmd, split)
+	elif subcmd == 'sell':
+		_stalk_sales(cmd, split)
+
+def _stalk_sales(cmd, split):
+	if len(split) != 2:
+		_stalk_list_sale_prices(cmd)
+		return
+	user_id = cmd.sender['id']
+	current_time = datetime.now(timezone.utc)
+	value = int(split[1])
+	try:
+		with db:
+			db.execute('''
+			INSERT INTO sale_price VALUES (?, ?, ?)
+			''', (user_id, current_time, value))
+		cmd.reply('Sale price recorded!')
+	except sqlite3.IntegrityError:
+		cmd.reply('Could not add sale price. Have you registered a friend code?')
+
+def _stalk_list_sale_prices(cmd):
+	# TODO: is there a better way to do this without a gross global?
+	global cached_row_id
+	cur = db.execute('''
+	SELECT sale_price.rowid, *
+	FROM sale_price
+	INNER JOIN user ON sale_price.user_id = user.id
+	WHERE sale_price.rowid > ?
+	''', (cached_row_id,))
+
+	prices = cur.fetchall()
+	current_time = datetime.now(timezone.utc)
+	results = {}
+	for price in prices:
+		user_id = price['user_id']
+
+		if dateutil.parser.parse(price['created_at']).date() != current_time.date():
+			cached_row_id = price['rowid']
+			continue
```

Comment:
without an `ORDER BY`, it seems very dangerous to assume you're getting the oldest rows first